interfaces/silas_ui.tcl

```diff
@@ -0,0 +1,395 @@
+
+proc iconik_wakeup {} {
+	set_next_page "off" "off"
+	page_show "off"
+	start_idle
+}
+
+proc iconik_water_temperature {} {
+	if {$::settings(enable_fahrenheit) == 1} {
+		set temp [round_to_one_digits [celsius_to_fahrenheit $::iconik_settings(water_temperature_overwride)]]
+		return "$temp F"
+	}
+	set temp [round_to_one_digits $::iconik_settings(water_temperature_overwride)]
+	return "$temp °C"
+}
+
+proc profile_file {} {
+	return "[homedir]/profiles/${::settings(profile_filename)}.tcl"
+}
+
+proc profile_backup_file {} {
+	return "[profile_file].orig"
+}
+
+proc profile_backup_exists {} {
+	set origfn [profile_backup_file]
+	return [file exists $origfn]
+}
+
+proc backup_profile {} {
+	if {![profile_backup_exists]} {
+		file copy -force [profile_file] [profile_backup_file]
+		borg toast [translate "Original profile backed up"]
+	}
+}
+
+proc is_advanced_profile {} {
+	return [expr {$::settings(settings_profile_type) == "settings_2c2" || $::settings(settings_profile_type) == "settings_2c"}]
+}
+
+proc restore_profile {} {
+	if {[profile_backup_exists]} {
+		file copy -force [profile_backup_file] [profile_file] 
+		file delete [profile_backup_file]
+		
+		select_profile $::settings(profile)
```

Comment:
i think loading a profile of the same name over does not yield the result expected. I could crash the app by converting the profile to an advanced profile, saving and reverting

Reply:
Hmm ok, for me it works so far. But this is the feature I use least often, so I might just be lucky. What would be a good solution? Loading another profile as a workaround in between? Feels not so clean :/